spec/spec.md

```diff
@@ -0,0 +1,180 @@
+exceptions, with the following exceptions:
+
+- Exception files must be placed in the `LICENSES/` directory in the root of the
+  project.
+
+- The tags `Valid-Exception-Identifier` and `Exception-Text` must be used
+  instead of `Valid-License-Identifier` and `License-Text`.
+
+<!-- Can we use LicenseRef for exceptions? -->
+
+<!-- TODO: Change "copyright and licensing information" to something shorter?
+Shorten it to CALI? -->
+
+## Copyright and licensing information
+
+Each file in the project must have copyright and licensing information
+associated with it, except the following files:
+
+- The license files.
+
+- The files belonging to the project's version control system (example:
+  `.git/`).
+
+- The files ignored by the version control system (example: `.gitignore`).
+
+- The files in the `.reuse/` directory in the root of the project.
+
+<!-- TODO: The files inside of a git submodule -->
+
+There are various ways to associate copyright and licensing information with a
+file.
+
+### Comment headers
+
+Each plain text file that can contain comments should contain comments at the
+top of the file (comment header) that declare that file's copyright and
+licensing information.
+
+If a file is not a plain text file or does not permit the inclusion of
+comments, the comment header that declares the file's copyright and licensing
+information should be in an adjacent file of the same name with the
+additional extension `.license` (example: `cat.jpg.license` if the original
+file is `cat.jpg`).
+
+The comment header must contain one or more `SPDX-Copyright` tags, and one or
+more `SPDX-License-Information` tags. A tag is followed by a colon, followed by
+a text value, and terminated by a newline.
+
+The `SPDX-Copyright` tag must be followed by a copyright notice as described
+in <!-- FIXME: Refer to an internal section -->.
+
+Instead of the `SPDX-Copyright` tag, the symbols `Copyright`, `Copyright (c)`,
+`Copyright (C)`, `Copyright ©` or `©` may be used.
+
+The `SPDX-License-Identifier` tag must be followed by a valid SPDX License
+Expression describing the licensing of the file (example:
+`SPDX-License-Identifier: GPL-3.0-or-later OR Apache-2.0`). If separate sections
+of the file are licensed differently, a different `SPDX-License-Identifier` tag
+should be included for each section.
+
+An example of a comment header:
+
+```
+# SPDX-Copyright: 2016, 2018-2019 Jane Doe <jane@example.com>
+# SPDX-Copyright: 2019 Example Company
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+```
+
+### DEP5
+
+Copyright and licensing information may be associated with a file through a DEP5
+file. The intended use case of this method is large directories where including
+a comment header in each file (or in `.license` companion files) is impossible
+or undesirable.
+
+The DEP5 file must be in `.reuse/dep5` from the root of the project.
+
+<!-- TODO: How to do multiple licenses without AND and OR. -->
```

Comment:
DEP5 does make use both of `and` and `or` as well as brackets `(` and `)` for grouping – identical to SPDX syntax.

There are some other incompatibilities between DEP5 and SPDX though, mainly in the names of some of the licenses:
https://wiki.debian.org/Proposals/CopyrightFormat#Differences_between_DEP5_and_SPDX

Reply:
I know, but I'm trying to do the following:

```
SPDX-License-Identifier: MIT
SPDX-License-Identifier: 0BSD
```

This is different from `MIT AND 0BSD` and `MIT OR 0BSD`. It means that part of the file is under MIT, and part is under 0BSD.

`MIT AND 0BSD` suggests that both licenses apply to the contents of the file and that they must both be obeyed, which is really really close.

`MIT OR 0BSD` suggests that the user may choose which license applies to them, which is not correct.

Reply:
Oh, as addendum, another way that the SPDX spec deals with this is:

```
SPDX-License-Identifier: MIT; 0BSD
```

(but the tool currently does not support this :x)

Reply:
Could you just list separate licenses on separate SPDX-License-Identifier lines as described above?  In converting to an SPDX document, the tools which parse the source files would need to convert them to an expression for the file declared license, but it could also add them individually to the `PackageLicenseInfoFromFiles` which is a list of licenses collected from the source files.

Reply:
That's what we normally do in source code files, but that isn't supported inside of the DEP5 spec, that I know of.

It's a bit of a pedantic thing, though, because the operators `AND` and `;` are _almost_ identical.

Reply:
Coming back to this: I suggested that the SPDX spec allows an expression such as

```
SPDX-License-Identifier: MIT; 0BSD
```

This is false. I researched this further today, and could not find it in the spec. Instead, I had previously found it [here](https://wiki.spdx.org/view/FileNoticeExamples#Package_Examples), but had not read the disclaimer that the spec does not (yet?) support this.

Reply:
I can confirm that most license expression parsers don't support this operator.

In any case, I would recommend not using the `;` as most SPDX license expression parsers will fail and there is no agreed to definition in the SPDX spec.

I added an issue to the SPDX Spec to track either adding to to the spec or removing it from the wiki page to avoid future confusion: https://github.com/spdx/spdx-spec/issues/123